server/src/main/java/org/opensearch/search/startree/filter/provider/DimensionFilterMapper.java

```diff
@@ -0,0 +1,385 @@
+    long convertToDocValues(Number parsedValue) {
+        return NumericUtils.floatToSortableInt((Float) parsedValue);
+    }
+
+    @Override
+    Number getNextLow(Number parsedValue) {
+        return FloatPoint.nextDown((Float) parsedValue);
+    }
+
+    @Override
+    Number getNextHigh(Number parsedValue) {
+        return FloatPoint.nextUp((Float) parsedValue);
+    }
+}
+
+class DoubleFieldMapperNumeric extends NumericDecimalFieldMapper {
+    @Override
+    long convertToDocValues(Number parsedValue) {
+        return NumericUtils.doubleToSortableLong((Double) parsedValue);
+    }
+
+    @Override
+    Number getNextLow(Number parsedValue) {
+        return DoublePoint.nextDown((Double) parsedValue);
+    }
+
+    @Override
+    Number getNextHigh(Number parsedValue) {
+        return DoublePoint.nextUp((Double) parsedValue);
+    }
+}
+
+class KeywordFieldMapper implements DimensionFilterMapper {
+
+    @Override
+    public DimensionFilter getExactMatchFilter(MappedFieldType mappedFieldType, List<Object> rawValues) {
+        if (mappedFieldType instanceof KeywordFieldType) {
+            KeywordFieldType keywordFieldType = (KeywordFieldType) mappedFieldType;
+            List<Object> convertedValues = new ArrayList<>(rawValues.size());
+            for (Object rawValue : rawValues) {
+                convertedValues.add(parseRawKeyword(mappedFieldType.name(), rawValue, keywordFieldType));
+            }
+            return new ExactMatchDimFilter(mappedFieldType.name(), convertedValues);
+        }
+        return null;
+    }
+
+    @Override
+    public DimensionFilter getRangeMatchFilter(
+        MappedFieldType mappedFieldType,
+        Object rawLow,
+        Object rawHigh,
+        boolean includeLow,
+        boolean includeHigh
+    ) {
+        if (mappedFieldType instanceof KeywordFieldType) {
+            KeywordFieldType keywordFieldType = (KeywordFieldType) mappedFieldType;
+            return new RangeMatchDimFilter(
+                mappedFieldType.name(),
+                parseRawKeyword(mappedFieldType.name(), rawLow, keywordFieldType),
+                parseRawKeyword(mappedFieldType.name(), rawHigh, keywordFieldType),
+                includeLow,
+                includeHigh
+            );
+        }
+        return null;
+    }
+
+    @Override
+    public Optional<Long> getMatchingOrdinal(
+        String dimensionName,
+        Object value,
+        StarTreeValues starTreeValues,
+        DimensionFilter.MatchType matchType
+    ) {
+        StarTreeValuesIterator genericIterator = starTreeValues.getDimensionValuesIterator(dimensionName);
+        if (genericIterator instanceof SortedSetStarTreeValuesIterator) {
+            SortedSetStarTreeValuesIterator sortedSetIterator = (SortedSetStarTreeValuesIterator) genericIterator;
+            try {
+                if (matchType == DimensionFilter.MatchType.EXACT) {
+                    return Optional.of(sortedSetIterator.lookupTerm((BytesRef) value));
```

Comment:
is there a reason to not return empty if we don't have a match ?

Reply:
Good catch.
Added the same